src/mcts/stoppers/alphazero.cc

```diff
@@ -74,7 +71,7 @@ std::unique_ptr<SearchStopper> AlphazeroTimeManager::GetStopper(
   auto total_moves_time = *time + increment - move_overhead_;
   
   // use the increment in the first upcoming move
-  float this_move_time = increment + (total_moves_time / alphazerotimevalue_);
+  float this_move_time = increment + (total_moves_time / (100.0f / alphazerotimepct_));
```

Comment:
You shouldn't use increment here. I think this should just be:
```
float this_move_time = (*time - move_overhead_) * alphazerotimepct_ / 100.0f
```

Reply:
Let me get to the bottom of this.

I think we do need the increase see screenshot below.. 

Of course, alphazero just used a budget and did not have an inrease per move, but if we have increase per move we should add it to the budgeted time for the upcoming move immediately. Below is the sheet I made to test out the difference. The blue line stalls at 240 seconds, because 5% of 240 seconds is the 12 seconds increase, which are budgeted and used and then next move again 12 seconds is added, therefore the line does not get below the 240 seconds (asymptote).

The red line is what we want to achieve: even if there is an increase we should use it.

Think it's correct but if you can point out a possible mistake please do :)

![image](https://user-images.githubusercontent.com/506830/82645536-4c802680-9c13-11ea-8a8b-f55b1e8327b8.png)

Reply:
You add the increment before the move, which is not correct. So at the first move the total time in your example is 300 seconds, not 312. (This is a quick comment, I'm a bit busy right now, will take a deeper look later).

Reply:
It looks as if the Inc is added twice and that looks strange  but it works out when UCI comes in with

Wtime 300000 ms winc 12000 ms.

Total time = wtime + winc

Move time = winc + (totaltime * perc)

Totaltime also includes the winc, but otherwise we don't use all possible given milliseconds.

Reply:
Think of an edge case, e.g. 100% perc, no move overhead. Then for the first move you will have
```
Total time = 312
Move time = 324
```
I agree that the behavior with percentages that give a move time under the increment is strange, but I'm not sure this needs fixing.

Reply:
Hmm, when somebody would budget close to all time to the first move then the graph begins to look strange anyways indeed.

Here is an example for 75%.

![image](https://user-images.githubusercontent.com/506830/82668310-c3301a80-9c39-11ea-98ad-5a061233768e.png)

Basically I think when an increment is set, that increment should be used, as is the case with the current implementation. While if during testing the clock would not go below the threshhold of increment+chosen percentage (like in this example go from 16 to 4 seconds and thus total time stays at16 seconds) that gives a wrong impression. Therefore I would prefer to keep the calculation as it is now.